### ISSUE TITLE
Introduce `context-set:`

### ISSUE THREAD

**Issue by ReinhardKeil** [Open-CMSIS-Pack/devtools#896]
When working with multiple projects it is required to know which `-context` selection should be used to debug and download to the target hardware.

The proposed [`context-set:`](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/projmgr/docs/Manual/YML-CBuild-Format.md#context-set) delivers this information.

To make it useful it requires that more then one `--context` can be specified on the command-line of cbuild and csolution. See: #854

**Comment by jkrech** [comment]
@mcgordonite this sounds like a valid conflict.
Prior to check-in of the cbuild files the user must have run a complete convert to have the full data for reproducing.

For retaining information about context-sets I would prefer to define this information in a persistent input file like csolution.yml.

```
  context-sets:
    - <context-setA>: <context1>, <context2>, <context3>
    - <context-setB>: <context2b>, <context3>
```

cbuild and csolution would accept the name of the context-set at the command line instead of using `--context`.

**Comment by ReinhardKeil** [comment]
@mcgordonite thanks for highlighting. I agree this is an issue and we should keep the information about build information and context sets in different files for that reason. I requires further discussion.

**Comment by ReinhardKeil** [comment]
This issue needs review and an different proposal.  We need a file that stores which context settings are currently created for an image.

**Comment by brondani** [comment]
The following diagram shows an example about how context build-runs would be triggered depending on different `context-set`:

![context-set](https://github.com/Open-CMSIS-Pack/devtools/assets/12509084/44465e4b-cc4c-4e8e-bbaa-fe571174164e)

The output directory and artifact names should **not** be reused by different contexts because:
- it intrudes into the build system: to implement scenario 4 the default build dependencies would have to be skipped.
- it breaks predictability: what build configuration relates to a given build artifact on disk?
- it requires a rebuild when changing configuration.
 
In my opinion the `context-set` should be handled as a descriptive **input** yml node, directly or indirectly pointing to its relevant build artifacts. Artifacts should never be overwritten on disk by different context build-runs. If a context-set common output directory is still desired a `context-set output` node could be introduced or a post-build step could be used to copy the artifacts into a given location.
 
In practice I see `context-set` as an indication about what artifacts should be considered for loading/debugging, so triggering building but without intervening in the building mechanism itself. In other words choosing a `context-set` should not change the build instructions (cbuild-idx.yml and cbuild.yml files).

**Comment by ReinhardKeil** [comment]
This is implemented sufficiently in CMSIS-Toolbox 2.0. Further enhancements are planned, but need to be scoped out. Hence I'm closing this.
